### PR TITLE
boards/esp32: changes the approach for configurations of DAC channels in board definitions

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR changes the approach of peripheral configurations for DAC channels in board definitions to the usual RIOT approach. With these changes, peripheral configurations use static const arrays in the `boards/esp32*/periph_conf.h` files and define the `*_NUMOF` macros using the size of these static array.

The static configuration arrays contain only definitions that can be changed by the board definition or the application. They do not contain any MCU implementation detail. The board definitions use preprocessor defines as before to fill these static configuration arrays. This makes it possible to override all configurations either with the make command or application specific configuration files.

Please note that commit https://github.com/RIOT-OS/RIOT/commit/8b48dfd62b9ef74dcf3bf023d2fe30fefe76dee3 is in also in related PRs to get each PR compilable separately.

### Testing procedure

Compilation and test with the most common ESP32 board should be executed
```
make BOARD=esp32-wroom-32 -C tests/periph_dac flash test
```

### Issues/PRs references

PRs #11289 #11290 #11291 #11292 #11293 #11294 are releated and should be merged together.
Depends on PR #11289.

### CODE REVIEW

File: cpu/esp32/periph/adc.c [hunk 1]
```diff
@@ -397,6 +397,7 @@ static void _adc_module_init(void)
 
 #if defined(DAC_GPIOS)
 
+static const unsigned dac_chn_num = DAC_NUMOF;
```
Comment (leandrolanzieri) [inline cpu/esp32/periph/adc.c]: `dac_chn_num` is being define two times, also, couldn't `DAC_NUMOF` be used instead?

File: boards/esp32-olimex-evb/Makefile.features [hunk 2]
```diff
@@ -2,8 +2,6 @@
 include $(RIOTBOARD)/common/esp32/Makefile.features
 
 # additional features provided by the board (no ADC and no DAC)
-FEATURES_PROVIDED += periph_adc
-FEATURES_PROVIDED += periph_dac
```
Comment (leandrolanzieri) [inline boards/esp32-olimex-evb/Makefile.features]: Maybe move this to an own commit?

**Comment by leandrolanzieri** [comment]
Tested on esp32-wroom-32 and works fine. I would still move the esp32-olimex-evb change to a different commit as it actually is not so related to the change of the way it is configured.

File: boards/esp32-olimex-evb/Makefile.features (hunk 2 above)
Comment (gschorcht) [reply boards/esp32-olimex-evb/Makefile.features]: Done

**Comment by gschorcht** [comment]
@leandrolanzieri I reverted the commit with `esp32-olimex-evb` and splitted it.

**Comment by leandrolanzieri** [comment]
@gschorcht please squash

**Comment by gschorcht** [comment]
@leandrolanzieri Thanks for reviewing and testing. Squashed.

**Comment by leandrolanzieri** [comment]
> Need to retest after review

I meant rebase >:(

**Comment by gschorcht** [comment]
@leandrolanzieri All PRs are rebased. I'm not sure about the order in which we should merge them. All PRs are compilable and are working separatly. However, all of them require the changes in commit 8ede5bec1431b55f7ef5631bda16b38297f5cfc5 and include this commit therefore.

Therefore, I would suggest to start with PR #11289. After that, I will rebase all of them again.

**Comment by gschorcht** [comment]
@leandrolanzieri I rebased PRs #11290 to #11294.

**Comment by gschorcht** [comment]
@leandrolanzieri Thanks.
